tests/fediverse/follow/__init__.py

```diff
@@ -29,14 +29,14 @@ def get_collections(self):
 
 
     def test_unrelated(self):
-        self.leader_node.assert_not_member_of_collection_at(self.follower_actor_uri, self.leader_actor_followers_collection_uri)
-        self.leader_node.assert_not_member_of_collection_at(self.follower_actor_uri, self.leader_actor_following_collection_uri)
-        self.follower_node.assert_not_member_of_collection_at(self.leader_actor_uri, self.follower_actor_followers_collection_uri)
-        self.follower_node.assert_not_member_of_collection_at(self.leader_actor_uri, self.follower_actor_following_collection_uri)
+        self.leader_node.hard_assert_not_member_of_collection_at(self.follower_actor_uri, self.leader_actor_followers_collection_uri)
```

Comment:
I prefer considering `assert_that` to be "hard" by default and creating a `soft_assert_that` for the special cases. However, I'm not exactly sure how the soft assert will be used and how it will be presented in reports, so I'm keeping an open mind on the topic.

Reply:
Hamcrest's `assert_that` raises the default Python `AssertionError`, while obviously our assertions need to raise a different `Exception`. I wanted all those to have a common supertype, so that's where that comes from. Also lets us sort thrown exceptions by whether they come from explicit assertions in the test, or some Python `assert` somewhere else in the codebase. Doesn't have to be that way of course, I guess we can revisit this any time.